code/modules/antagonists/heretic/heretic_antag.dm

```diff
@@ -84,33 +84,99 @@
 	LAZYNULL(sac_targets)
 	return ..()
 
+/datum/antagonist/heretic/proc/get_icon_of_knowledge(datum/heretic_knowledge/knowledge)
+	if(ispath(knowledge))
+		//if the argument is a typepath, we need to make a dummy so we can access some of its properties
+		knowledge = new knowledge()
+
+	//if the knowledge has a special icon, use that
+	if(!isnull(knowledge.research_tree_icon))
+		return knowledge.research_tree_icon
+
+	//if the knowledge is a transmutation, use the result
+	if(!isnull(knowledge.result_atoms) && knowledge.result_atoms.len)
+		var/atom/result_thing = knowledge.result_atoms[1]
+		return icon(result_thing.icon,result_thing.icon_state,frame=1)
```

Comment:
All these icons should ideally be generated once, and then we can quickly reference the icon rather than needing to re-create it

Reply:
so, I imagine calling icon() during ui_static_data should do the trick, right?

Reply:
Better, but could be even better. Ideally we only ever call `icon` once, for every heretic in existence. We create a map of `knowledge typepath` to `generated icon` that we draw from instead of re-creating it whenever a heretic opens their UI.

Reply:
Wait I'm stupid. You should be making a spritesheet for this. That's the correct way to go.